discovery_engine_core/web-api/src/models.rs

```diff
@@ -74,24 +74,37 @@ impl AiDocument for PersonalizedDocument {
     }
 }
 
-/// Represents a document sent for ingestion.
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub(crate) struct IngestedDocument {
-    /// Unique identifier of the document.
-    pub(crate) id: String,
+/// Arbitrary properties that can be attached to a document.
+pub type DocumentProperties = HashMap<String, serde_json::Value>;
 
-    /// Snippet used to calculate embeddings for a document.
-    pub(crate) snippet: String,
+/// Represents personalized documents query params.
+#[derive(Debug, Clone, Deserialize)]
+pub(crate) struct PersonalizedDocumentsQuery {
+    #[serde(deserialize_with = "deserialize_count_min_max")]
+    pub(crate) count: usize,
+}
 
-    /// Contents of the document properties.
-    pub(crate) properties: DocumentProperties,
+fn deserialize_count_min_max<'de, D>(deserializer: D) -> Result<usize, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let val = usize::deserialize(deserializer)?;
+    if (1_usize..=100_usize).contains(&val) {
+        Ok(val)
+    } else {
+        Err(de::Error::custom("count needs to be between 1 and 100"))
+    }
```

Comment:
❔ could this also be done via a filter? like [this example](https://github.com/seanmonstar/warp/blob/514af163f4f011c2d371cb207cfe3ccf43016abf/examples/rejections.rs) shows ?

Reply:
please check again :) and feel free to merge it if it's ok